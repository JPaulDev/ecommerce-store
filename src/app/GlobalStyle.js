import { createGlobalStyle } from 'styled-components';
import openSansRegular from '../assets/fonts/OpenSans-Regular.woff';
import openSansMedium from '../assets/fonts/OpenSans-Medium.ttf';
import openSansSemiBold from '../assets/fonts/OpenSans-SemiBold.woff';
import openSansBold from '../assets/fonts/OpenSans-Bold.woff';
import uniSansBook from '../assets/fonts/UniSans-Book.woff2';
import uniSansRegular from '../assets/fonts/UniSans-Regular.woff2';
import uniSansSemiBold from '../assets/fonts/UniSans-SemiBold.woff2';
import uniSansBold from '../assets/fonts/UniSans-Bold.woff';

const GlobalStyle = createGlobalStyle`
  :root {
    --color-orange-1: #ff9110;
    --color-grey-1: #434343;
    --color-grey-2: #333;
    --color-grey-3: #b3b3b3;
    --color-grey-4: #2f2f2f;
    --color-grey-5: #7e7e7e;
    --color-grey-6: #f6f6f6;
    --color-grey-7: #9e9e9e;
    --color-grey-8: #5b5b5b;
    --color-grey-9: #e1e1e1;
    --color-grey-10: #bcbec0;
    --color-grey-11: #cfcfcf;
    --color-grey-12: #c7c7c7;
    --color-blue-1: #3d70b7;
    --color-blue-2: #1766a6;
  }

  * {
    font-family: 'Open Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI',
      'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
      'Helvetica Neue', sans-serif;
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    border: 0;
  }

  html {
    font-size: 16px;
  }

  #root {
    position: relative;
    min-height: 100vh;
    overflow: clip;
  }

  @font-face {
    font-family: 'Open Sans';
    src: url(${openSansRegular}) format('woff');
    font-weight: 400;
  }

  @font-face {
    font-family: 'Open Sans';
    src: url(${openSansMedium}) format('truetype');
    font-weight: 500;
  }

  @font-face {
    font-family: 'Open Sans';
    src: url(${openSansSemiBold}) format('woff');
    font-weight: 600;
  }

  @font-face {
    font-family: 'Open Sans';
    src: url(${openSansBold}) format('woff');
    font-weight: 700;
  }

  @font-face {
    font-family: 'Uni Sans Book';
    src: url(${uniSansBook}) format('woff2');
  }

  @font-face {
    font-family: 'Uni Sans';
    src: url(${uniSansRegular}) format('woff2');
    font-weight: 400;
  }

  @font-face {
    font-family: 'Uni Sans';
    src: url(${uniSansSemiBold}) format('woff2');
    font-weight: 600;
  }

  @font-face {
    font-family: 'Uni Sans';
    src: url(${uniSansBold}) format('woff');
    font-weight: 700;
  }
`;

export default GlobalStyle;
