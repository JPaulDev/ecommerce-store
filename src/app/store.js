import { combineReducers, configureStore } from '@reduxjs/toolkit';
import basketReducer from '../features/basket/basketSlice';

const rootReducer = combineReducers({
  basket: basketReducer,
});

export const setupStore = (preloadedState) =>
  configureStore({
    reducer: rootReducer,
    preloadedState,
  });
