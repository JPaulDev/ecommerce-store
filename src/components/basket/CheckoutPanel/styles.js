import styled, { css } from 'styled-components';

const borderStyles = css`
  border-left: 1px solid var(--color-grey-9);
  border-right: 1px solid var(--color-grey-9);
`;

const flexStyles = css`
  align-items: center;
  display: flex;
`;

const linkStyles = css`
  font-weight: 700;
  text-decoration: none;
  text-transform: uppercase;
`;

export const TopContainer = styled.div`
  background-color: #d7f0db;
  border-top: 1px solid var(--color-grey-9);
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  padding: 20px 30px;
  ${borderStyles}
`;

export const Heading = styled.h2`
  border-bottom: 1px solid var(--color-grey-10);
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.4rem;
  font-weight: 600;
  margin-bottom: 15px;
  padding: 11px 0;

  @media screen and (max-width: 899px) {
    text-align: center;
  }
`;

export const Container = styled.div`
  column-gap: 20px;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 5px 0;
  ${flexStyles}

  strong {
    font-size: 1.25rem;
    margin-right: 5px;
  }
`;

export const Text = styled.span`
  font-size: ${({ $fontSize }) => $fontSize || '0.82rem'};
`;

export const InnerContainer = styled.div`
  border-top: 1px solid var(--color-grey-10);
  margin-top: 15px;
  padding: 20px 0 10px;
`;

export const CheckoutLink = styled.a`
  background-image: linear-gradient(to top, #009245, #02ab52);
  border-radius: 6px;
  box-shadow: var(--shadow-button);
  color: white;
  column-gap: 10px;
  font-size: 0.88rem;
  justify-content: center;
  margin: 0 auto 20px;
  padding: 15px;
  ${flexStyles}
  ${linkStyles}

  &:hover {
    background-image: linear-gradient(to top, #008631, #00a13f);
    box-shadow: var(--shadow-button-hover);
  }
`;

export const LoginLink = styled.a`
  color: var(--color-blue-1);
  display: block;
  font-size: 0.7rem;
  text-align: center;
  ${linkStyles}

  &:hover {
    text-decoration: underline;
  }
`;

export const BottomContainer = styled.div`
  background-color: #cadfce;
  border-bottom: 1px solid var(--color-grey-9);
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  flex-direction: column;
  padding: 20px 30px 30px;
  ${borderStyles}
  ${flexStyles}

  img {
    margin-top: 5px;
  }
`;

export const PaypalLink = styled.a`
  background-color: #ffc439;
  border-radius: 100px;
  display: flex;
  margin: 5px 0;
  padding: 10px 25px;

  &:hover {
    box-shadow: inset 0 0 100px 100px rgb(0 0 0 / 5%);
  }
`;
