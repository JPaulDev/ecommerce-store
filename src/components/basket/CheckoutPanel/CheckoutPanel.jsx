import Image from 'next/future/image';
import Link from 'next/link';
import { useSelector } from 'react-redux';
import amazonPay from '../../../../public/images/basket/amazon-pay.webp';
import { selectBasketSubtotal } from '../../../features/basket/basketSlice';
import usePrice from '../../../lib/hooks/usePrice';
import { PayPal, Lock } from '../../icons';
import { Price } from '../../common';
import * as Styled from './styles';

export default function CheckoutPanel() {
  const subtotal = useSelector(selectBasketSubtotal);
  const delivery = 4.99;

  const { price } = usePrice({
    price: subtotal + delivery,
    currencyCode: 'GBP',
  });

  return (
    <div>
      <Styled.TopContainer>
        <Styled.Heading>Basket Summary</Styled.Heading>
        <Styled.Container>
          <Styled.Text>Delivery to UK Mainland</Styled.Text>
          <Price
            price="Â£4.99"
            fontSize="var(--font-size-20)"
            fontWeight="var(--font-weight-medium)"
          />
        </Styled.Container>
        <Styled.Container>
          <span>
            <strong>Total</strong>
            (including VAT)
          </span>
          <Price
            price={price}
            fontSize="var(--font-size-26)"
            fontWeight="var(--font-weight-medium)"
          />
        </Styled.Container>
        <Styled.Text>Finance options available at checkout</Styled.Text>
        <Styled.InnerContainer>
          <Link href="/" passHref>
            <Styled.CheckoutLink>
              Checkout As Guest
              <Lock width={15} height={17} />
            </Styled.CheckoutLink>
          </Link>
          <Link href="/account" passHref>
            <Styled.LoginLink>
              Have an account? Login and checkout now
            </Styled.LoginLink>
          </Link>
        </Styled.InnerContainer>
      </Styled.TopContainer>
      <Styled.BottomContainer>
        <Styled.Text>Alternatively</Styled.Text>
        <Link href="https://www.paypal.co.uk" passHref>
          <Styled.PaypalLink
            title="Pay with PayPal"
            aria-label="Pay with PayPal"
          >
            <PayPal width={80} height={20} />
          </Styled.PaypalLink>
        </Link>
        <Styled.Text>Two easy ways to pay</Styled.Text>
        <Link href="https://www.amazon.co.uk" passHref>
          <a>
            <Image
              src={amazonPay}
              title="Pay with Amazon Pay"
              alt="Pay with Amazon Pay."
            />
          </a>
        </Link>
      </Styled.BottomContainer>
    </div>
  );
}
