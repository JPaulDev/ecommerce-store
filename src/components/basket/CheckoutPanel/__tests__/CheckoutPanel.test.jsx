import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { setupStore } from '../../../../app/store';
import CheckoutPanel from '../index';

describe('CheckoutPanel component', () => {
  it('renders a checkout panel with a checkout button and the basket total', () => {
    const tree = renderer
      .create(
        <Provider store={setupStore()}>
          <CheckoutPanel />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
