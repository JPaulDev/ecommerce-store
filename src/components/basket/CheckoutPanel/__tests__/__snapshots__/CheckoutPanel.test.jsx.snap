// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPanel component renders a checkout panel with a checkout button and the basket total 1`] = `
.c4 {
  color: var(--color-blue-1);
  font-size: 1.3rem;
  font-weight: 500;
}

.c6 {
  color: var(--color-blue-1);
  font-size: 1.6rem;
  font-weight: 500;
}

.c5 {
  font-size: 0.7em;
}

.c0 {
  background-color: #d7f0db;
  border-top: 1px solid var(--color-grey-9);
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  padding: 20px 30px;
  border-left: 1px solid var(--color-grey-9);
  border-right: 1px solid var(--color-grey-9);
}

.c1 {
  border-bottom: 1px solid var(--color-grey-10);
  font-family: 'Uni Sans',sans-serif;
  font-size: 1.4rem;
  font-weight: 600;
  margin-bottom: 15px;
  padding: 11px 0;
}

.c2 {
  -webkit-column-gap: 20px;
  column-gap: 20px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 5px 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 strong {
  font-size: 1.25rem;
  margin-right: 5px;
}

.c3 {
  font-size: 0.9rem;
}

.c7 {
  font-size: 0.82rem;
}

.c8 {
  border-top: 1px solid var(--color-grey-10);
  margin-top: 15px;
  padding: 20px 0 10px;
}

.c9 {
  background-image: linear-gradient(to top,#009245,#02ab52);
  border-radius: 6px;
  box-shadow: var(--shadow-button);
  color: white;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  font-size: 0.88rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 20px;
  padding: 15px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
}

.c9:hover {
  background-image: linear-gradient(to top,#008631,#00a13f);
  box-shadow: var(--shadow-button-hover);
}

.c10 {
  color: var(--color-blue-1);
  display: block;
  font-size: 0.7rem;
  text-align: center;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
}

.c10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11 {
  background-color: #cadfce;
  border-bottom: 1px solid var(--color-grey-9);
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px 30px 30px;
  border-left: 1px solid var(--color-grey-9);
  border-right: 1px solid var(--color-grey-9);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 img {
  margin-top: 5px;
}

.c12 {
  background-color: #ffc439;
  border-radius: 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 5px 0;
  padding: 10px 25px;
}

.c12:hover {
  box-shadow: inset 0 0 100px 100px rgb(0 0 0 / 5%);
}

@media screen and (max-width:899px) {
  .c1 {
    text-align: center;
  }
}

<div>
  <div
    className="c0"
  >
    <h2
      className="c1"
    >
      Basket Summary
    </h2>
    <div
      className="c2"
    >
      <span
        className="c3"
      >
        Delivery to UK Mainland
      </span>
      <span
        className="c4"
      >
        <span
          className="c5"
        >
          £
        </span>
        4
        <span
          className="c5"
        >
          .99
        </span>
      </span>
    </div>
    <div
      className="c2"
    >
      <span>
        <strong>
          Total
        </strong>
        (including VAT)
      </span>
      <span
        className="c6"
      >
        <span
          className="c5"
        >
          £
        </span>
        4
        <span
          className="c5"
        >
          .99
        </span>
      </span>
    </div>
    <span
      className="c7"
    >
      Finance options available at checkout
    </span>
    <div
      className="c8"
    >
      <a
        className="c9"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Checkout As Guest
        <svg
          fill="currentColor"
          height={17}
          viewBox="0 0 448 512"
          width={15}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M144 144v48H304V144c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192V144C80 64.5 144.5 0 224 0s144 64.5 144 144v48h16c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64H80z"
          />
        </svg>
      </a>
      <a
        className="c10"
        href="/account"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Have an account? Login and checkout now
      </a>
    </div>
  </div>
  <div
    className="c11"
  >
    <span
      className="c7"
    >
      Alternatively
    </span>
    <a
      aria-label="Pay with PayPal"
      className="c12"
      href="https://www.paypal.co.uk"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      title="Pay with PayPal"
    >
      <svg
        height={20}
        viewBox="-.02682843 0 123.63183286 30.17842908"
        width={80}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m46.211 6.749h-6.839a.95.95 0 0 0 -.939.802l-2.766 17.537a.57.57 0 0 0 .564.658h3.265a.95.95 0 0 0 .939-.803l.746-4.73a.95.95 0 0 1 .938-.803h2.165c4.505 0 7.105-2.18 7.784-6.5.306-1.89.013-3.375-.872-4.415-.972-1.142-2.696-1.746-4.985-1.746zm.789 6.405c-.374 2.454-2.249 2.454-4.062 2.454h-1.032l.724-4.583a.57.57 0 0 1 .563-.481h.473c1.235 0 2.4 0 3.002.704.359.42.469 1.044.332 1.906zm19.654-.079h-3.275a.57.57 0 0 0 -.563.481l-.145.916-.229-.332c-.709-1.029-2.29-1.373-3.868-1.373-3.619 0-6.71 2.741-7.312 6.586-.313 1.918.132 3.752 1.22 5.031.998 1.176 2.426 1.666 4.125 1.666 2.916 0 4.533-1.875 4.533-1.875l-.146.91a.57.57 0 0 0 .562.66h2.95a.95.95 0 0 0 .939-.803l1.77-11.209a.568.568 0 0 0 -.561-.658zm-4.565 6.374c-.316 1.871-1.801 3.127-3.695 3.127-.951 0-1.711-.305-2.199-.883-.484-.574-.668-1.391-.514-2.301.295-1.855 1.805-3.152 3.67-3.152.93 0 1.686.309 2.184.892.499.589.697 1.411.554 2.317zm22.007-6.374h-3.291a.954.954 0 0 0 -.787.417l-4.539 6.686-1.924-6.425a.953.953 0 0 0 -.912-.678h-3.234a.57.57 0 0 0 -.541.754l3.625 10.638-3.408 4.811a.57.57 0 0 0 .465.9h3.287a.949.949 0 0 0 .781-.408l10.946-15.8a.57.57 0 0 0 -.468-.895z"
          fill="#253b80"
        />
        <path
          d="m94.992 6.749h-6.84a.95.95 0 0 0 -.938.802l-2.766 17.537a.569.569 0 0 0 .562.658h3.51a.665.665 0 0 0 .656-.562l.785-4.971a.95.95 0 0 1 .938-.803h2.164c4.506 0 7.105-2.18 7.785-6.5.307-1.89.012-3.375-.873-4.415-.971-1.142-2.694-1.746-4.983-1.746zm.789 6.405c-.373 2.454-2.248 2.454-4.062 2.454h-1.031l.725-4.583a.568.568 0 0 1 .562-.481h.473c1.234 0 2.4 0 3.002.704.359.42.468 1.044.331 1.906zm19.653-.079h-3.273a.567.567 0 0 0 -.562.481l-.145.916-.23-.332c-.709-1.029-2.289-1.373-3.867-1.373-3.619 0-6.709 2.741-7.311 6.586-.312 1.918.131 3.752 1.219 5.031 1 1.176 2.426 1.666 4.125 1.666 2.916 0 4.533-1.875 4.533-1.875l-.146.91a.57.57 0 0 0 .564.66h2.949a.95.95 0 0 0 .938-.803l1.771-11.209a.571.571 0 0 0 -.565-.658zm-4.565 6.374c-.314 1.871-1.801 3.127-3.695 3.127-.949 0-1.711-.305-2.199-.883-.484-.574-.666-1.391-.514-2.301.297-1.855 1.805-3.152 3.67-3.152.93 0 1.686.309 2.184.892.501.589.699 1.411.554 2.317zm8.426-12.219-2.807 17.858a.569.569 0 0 0 .562.658h2.822c.469 0 .867-.34.939-.803l2.768-17.536a.57.57 0 0 0 -.562-.659h-3.16a.571.571 0 0 0 -.562.482z"
          fill="#179bd7"
        />
        <path
          d="m7.266 29.154.523-3.322-1.165-.027h-5.563l3.866-24.513a.316.316 0 0 1 .314-.268h9.38c3.114 0 5.263.648 6.385 1.927.526.6.861 1.227 1.023 1.917.17.724.173 1.589.007 2.644l-.012.077v.676l.526.298a3.69 3.69 0 0 1 1.065.812c.45.513.741 1.165.864 1.938.127.795.085 1.741-.123 2.812-.24 1.232-.628 2.305-1.152 3.183a6.547 6.547 0 0 1 -1.825 2c-.696.494-1.523.869-2.458 1.109-.906.236-1.939.355-3.072.355h-.73c-.522 0-1.029.188-1.427.525a2.21 2.21 0 0 0 -.744 1.328l-.055.299-.924 5.855-.042.215c-.011.068-.03.102-.058.125a.155.155 0 0 1 -.096.035z"
          fill="#253b80"
        />
        <path
          d="m23.048 7.667c-.028.179-.06.362-.096.55-1.237 6.351-5.469 8.545-10.874 8.545h-2.752c-.661 0-1.218.48-1.321 1.132l-1.409 8.936-.399 2.533a.704.704 0 0 0 .695.814h4.881c.578 0 1.069-.42 1.16-.99l.048-.248.919-5.832.059-.32c.09-.572.582-.992 1.16-.992h.73c4.729 0 8.431-1.92 9.513-7.476.452-2.321.218-4.259-.978-5.622a4.667 4.667 0 0 0 -1.336-1.03z"
          fill="#179bd7"
        />
        <path
          d="m21.754 7.151a9.757 9.757 0 0 0 -1.203-.267 15.284 15.284 0 0 0 -2.426-.177h-7.352a1.172 1.172 0 0 0 -1.159.992l-1.564 9.906-.045.289a1.336 1.336 0 0 1 1.321-1.132h2.752c5.405 0 9.637-2.195 10.874-8.545.037-.188.068-.371.096-.55a6.594 6.594 0 0 0 -1.017-.429 9.045 9.045 0 0 0 -.277-.087z"
          fill="#222d65"
        />
        <path
          d="m9.614 7.699a1.169 1.169 0 0 1 1.159-.991h7.352c.871 0 1.684.057 2.426.177a9.757 9.757 0 0 1 1.481.353c.365.121.704.264 1.017.429.368-2.347-.003-3.945-1.272-5.392-1.399-1.593-3.924-2.275-7.155-2.275h-9.38c-.66 0-1.223.48-1.325 1.133l-3.907 24.765a.806.806 0 0 0 .795.932h5.791l1.454-9.225z"
          fill="#253b80"
        />
      </svg>
    </a>
    <span
      className="c7"
    >
      Two easy ways to pay
    </span>
    <a
      href="https://www.amazon.co.uk"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt="Pay with Amazon Pay."
        data-nimg="future"
        decoding="async"
        height={24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        title="Pay with Amazon Pay"
        width={24}
      />
    </a>
  </div>
</div>
`;
